internal/nightfallconfig/nightfall_config.go

```diff
@@ -32,28 +39,40 @@ type Config struct {
 	FileExclusionList          []string
 }
 
-// GetNightfallConfigFile loads nightfall config from file
-func GetNightfallConfigFile(workspacePath, fileName string) (*NightfallConfigFileStructure, error) {
+// GetNightfallConfigFile loads nightfall config from file, returns default if missing/invalid
+func GetNightfallConfigFile(workspacePath, fileName string, logger logger.Logger) *NightfallConfigFileStructure {
+	defaultNightfallConfig := &NightfallConfigFileStructure{
+		Detectors:         []*nightfallAPI.Detector{&apiKeyDetector, &cryptoKeyDetector},
+		MaxNumberRoutines: DefaultMaxNumberRoutines,
+	}
 	nightfallConfigFile, err := os.Open(path.Join(workspacePath, fileName))
 	if err != nil {
-		return nil, err
+		logger.Warning(fmt.Sprintf("Error opening nightfall config: %s", err.Error()))
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
 	defer nightfallConfigFile.Close()
 	byteValue, err := ioutil.ReadAll(nightfallConfigFile)
 	if err != nil {
-		return nil, err
+		logger.Warning(fmt.Sprintf("Error reading nightfall config: %s", err.Error()))
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
 	var nightfallConfig NightfallConfigFileStructure
 	err = json.Unmarshal(byteValue, &nightfallConfig)
 	if err != nil {
-		return nil, err
+		logger.Warning("Invalid nightfall config")
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
 	if len(nightfallConfig.Detectors) < 1 {
-		return nil, errors.New("Nightfall config file is missing detectors")
+		logger.Warning("Nightfall config requires at least one detector")
+		logger.Info(defaultDetectorsInfoMessage)
+		return defaultNightfallConfig
 	}
-	if nightfallConfig.MaxNumberRoutines == 0 {
-		nightfallConfig.MaxNumberRoutines = defaultMaxNumberRoutines
+	if nightfallConfig.MaxNumberRoutines > MaxConcurrentRoutinesCap || nightfallConfig.MaxNumberRoutines <= 0 {
```

Comment:
hmm I think if someone specifies like 100 for the MaxNumberRoutines, we would want to give them as many as possible. If someone specifies 0 or less, we should give them the default. What do you think?

Reply:
Yeah I agree, I'll add an extra condition for that